### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by cedricbonhomme**
Try to fix some security related issues #1599 . 
Actually I didn't used mysql_real_escape_string() because it didn't worked on my own instance. Consequently I reverted it to addslashes() (https://github.com/cedricbonhomme/frontend/commit/65bb09e3c226412a77d2cc9468c1a183dfbf4535).
The error message was:
"""
Warning: mysql_real_escape_string() [function.mysql-real-escape-string]: Can't 
connect to local MySQL server through socket '/run/mysqld/mysqld.sock' (2) in 
/home/cedric/www/photos/src/libraries/adapters/DatabaseMySql.php on line 2661

Warning: mysql_real_escape_string() [function.mysql-real-escape-string]: A 
link to the server could not be established in /home/cedric/www/photos/src/
libraries/adapters/DatabaseMySql.php on line 2661
"""

cheers,

### CODE REVIEW

File: src/libraries/controllers/ApiUserController.php [hunk 1]
```diff
@@ -91,7 +91,7 @@ public function passwordReset()
     $password = $_POST['password'];
     $passwordConfirm = $_POST['password-confirm'];
     $tokenFromDb = $user->getAttribute('passwordToken');
-    if($tokenFromDb != $token)
+    if($tokenFromDb !== $token)
```
Comment (jmathai) [inline src/libraries/controllers/ApiUserController.php]: lgtm

File: src/libraries/controllers/ApiUserController.php [hunk 2]
```diff
@@ -55,7 +55,7 @@ public function passwordRequest()
     $email = $_POST['email'];
     if($email == $this->config->user->email)
     {
-      $token = md5(rand(10000,100000));
+      $token = md5(rand(10000,10000000));
```
Comment (jmathai) [inline src/libraries/controllers/ApiUserController.php]: lgtm

**Review by jmathai — COMMENTED** [review]
@cedricbonhomme I think the two lines where I commented `lgtm` are the ones needed for the patch - do you agree?

If so I will cherry pick just those lines out.

File: src/libraries/controllers/ApiUserController.php (hunk 2 above)
Comment (rverton) [reply src/libraries/controllers/ApiUserController.php]: I recommend using http://php.net/manual/de/function.openssl-random-pseudo-bytes.php for this.

**Comment by cedricbonhomme** [comment]
ok, thanks.
Maybe also, this one:

https://github.com/photo/frontend/pull/1600/commits/b7e927e6af676b5a9e0a5c8fb4e1a28f3e118525

**Comment by rverton** [comment]
Validation of the album parameter (https://github.com/photo/frontend/pull/1600/commits/e30e9a74b191224d8c7b390956b511024696c379#diff-00228b37eac6be321ea4cd2edece7d01R1860) is also crucial.

**Comment by jmathai** [comment]
Created pr-1601 with the appropriate commits cherry-picked. Will merge once tests pass.

**Comment by jmathai** [comment]
Merged #1601. Closing this PR. Thanks @rverton & @cedricbonhomme.

**Pull request closed without merging**